src/slskd/Integrations/Scripts/ScriptService.cs

```diff
@@ -64,45 +80,40 @@ private async Task HandleEvent(Event data)
             _ = Task.Run(() =>
             {
                 Process process = default;
+                var processId = Guid.NewGuid();
 
                 try
                 {
                     var run = script.Value.Run;
+                    string executable = run.Executable ?? DefaultExecutable;
+                    string args = run.Command;
 
-                    // split the string into at most 2 parts, leaving part [0] the executable and part [1] the rest of the string, but possibly null
-                    var parts = run.Split(" ", count: 2, options: StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.RemoveEmptyEntries);
-
-                    var executable = parts[0];
-                    string args = default;
-
-                    if (parts.Length > 1)
+                    if (string.IsNullOrEmpty(executable))
                     {
-                        args = parts[1].Replace("$DATA", data.ToJson());
+                        Log.Warning("Script '{Script}' will not be run: unable to determine script executable. Update the script configuration, or set your operating system's SHELL envar.");
+                        return;
                     }
 
-                    Log.Debug("Running script '{Script}': {Run}", script.Key, run);
+                    Log.Debug("Running script '{Script}': \"{Executable}\" {Args} (id: {ProcessId})", script.Key, executable, args, processId);
                     var sw = Stopwatch.StartNew();
 
                     process = new Process()
                     {
-                        StartInfo = new ProcessStartInfo(fileName: executable)
+                        StartInfo = new ProcessStartInfo(executable ?? DefaultExecutable)
                         {
+                            WorkingDirectory = Program.ScriptDirectory,
```

Comment:
What's the rationale for having a different WorkingDirectory?
I don't see any problem with it, but I also don't see any advantage

Reply:
creating any files from the command drops them in the executable's working directory, which is likely inside of a container, potentially with a read only filesystem, or maybe in a read only location on disk.  doing it this way will drop them into the user's /app directory, which is expected to be writable and is validated at startup.

mostly a 'pit of success' for less technical users